lib/components/MenuRenderer/MenuRenderer.tsx

```diff
@@ -137,7 +155,12 @@ export const MenuRenderer = ({
         case MENU_ITEM_ENTER:
         case MENU_ITEM_SPACE:
         case MENU_ITEM_CLICK: {
-          return { ...state, open: false, highlightIndex: CLOSED_INDEX };
+          return {
+            ...state,
+            ...('formElement' in action && action.formElement
+              ? null // Don't close the menu if the user clicked a "form element" item, e.g. checkbox
+              : { open: false, highlightIndex: CLOSED_INDEX }),
+          };
```

Comment:
Might be personal preference, but i would find this easier to read as an early return of `state`. 
```suggestion
          // Don't close the menu if the user clicked a "form element" item, e.g. checkbox
          if ('formElement' in action && action.formElement) {
            return state;
          }
          
          return { ...state, open: false, highlightIndex: CLOSED_INDEX };
```

Also given it's just returning `state` when it's a form element, should we not even [dispatch the item click](https://github.com/seek-oss/braid-design-system/pull/751/files#diff-b51a9060cc180c118af2f9b57a3c3eebR133) if it's a `formElement`?